### PR TITLE
Add plumbing to implement Neon instructions with wasm simd

### PR DESCRIPTION
**Pull request by seanptmaher**
I also added a few functions to test out the options -- tests work on my end. Let me know if anything's wrong here.

### CODE REVIEW

**Comment by seanptmaher** [comment]
The failed build I don't understand -- the errors seem to all be in x86 code, so I'm not sure what's wrong there. I'll look at it tomorrow.

**Comment by mr-c** [comment]
Hey @seanptmaher , it is us. Try working off of https://github.com/nemequ/simde/commit/7b3e46778edc39ac16658e79c5fabd3c0f59cbf5 until we fix the master branch

File: simde/arm/neon/int8x16.h [hunk 1]
```diff
@@ -116,6 +123,8 @@ simde_vmulq_s8(simde_int8x16_t a, simde_int8x16_t b) {
   simde_int8x16_t r;
 #if defined(SIMDE_NEON_NATIVE)
   r.n = vmulq_s8(a.n, b.n);
+#elif defined(SIMDE_NEON_WASM_SIMD128)
+  r.v128 = wasm_i8x16_mul(a.v128, b.v128);
```
Comment (mr-c) [inline simde/arm/neon/int8x16.h:127]: https://travis-ci.org/github/nemequ/simde/jobs/689963809#L569
``` make
../arm/neon/../../../simde/arm/neon/int8x16.h:127:12: error: implicit declaration of function 'wasm_i8x16_mul' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
  r.v128 = wasm_i8x16_mul(a.v128, b.v128);
           ^
```

**Comment by mr-c** [comment]
I fixed the master branch and merged it in ; so the issue I flagged is the only build failure remaining.

File: simde/arm/neon/int8x16.h (hunk 1 above)
Comment (seanptmaher) [reply simde/arm/neon/int8x16.h:127]: That's really weird, [3 days ago](https://github.com/llvm/llvm-project/commit/3181273be73ac798c3d5b9081dd53d87a31c91b3), they removed that function. I'll fix once I get back on my pc.

**Comment by nemequ** [comment]
If you can hold off on the implementations of NEON functions for a few days it should be in significantly better shape.  I've been working on a rewrite of the entire simde/arm directory which should be done very soon (@rosbif is waiting on this, too, so it's a high priority).

**Comment by ngzhian** [comment]
@nemequ How's the rewrite going? Please let us know if we can help in anyway. All the best, and have a good weekend!

**Comment by ngzhian** [comment]
Ignore me :) @seanptmaher pointed me to the gitter chat with your response.

**Comment by seanptmaher** [comment]
This is not really applicable anymore, closing.

**Pull request closed without merging**